src/server/pkg/storage/chunk/limit_objclient.go

```diff
@@ -0,0 +1,74 @@
+package chunk
+
+import (
+	"context"
+	io "io"
+	"math"
+
+	"github.com/pachyderm/pachyderm/src/server/pkg/obj"
+	"golang.org/x/sync/semaphore"
+)
+
+var _ obj.Client = limitedObjClient{}
+
+type limitedObjClient struct {
+	obj.Client
+	writersSem *semaphore.Weighted
```

Comment:
Those don't allow us to respect the context passed in to the Reader and Writer constructors.  It can't handle "I want a writer but I'm only willing to wait 1 minute".

Reply:
Fair point.